src/Microsoft.Extensions.CodeGeneration.Utils/DotNet/LibraryExportExtensions.cs

```diff
@@ -14,17 +14,28 @@ namespace Microsoft.Extensions.CodeGeneration.DotNet
 {
     public static class LibraryExportExtensions
     {
-        public static IEnumerable<MetadataReference> GetMetadataReferences(this LibraryExport export)
+        public static IEnumerable<MetadataReference> GetMetadataReferences(this LibraryExport export, bool throwOnError = true)
         {
             var references = new List<MetadataReference>();
             AssemblyMetadata assemblyMetadata;
             foreach (var lib in export.CompilationAssemblies)
             {
-                using (var stream = File.OpenRead(lib.ResolvedPath))
+                try
                 {
-                    var moduleMetadata = ModuleMetadata.CreateFromStream(stream, PEStreamOptions.PrefetchMetadata);
-                    assemblyMetadata = AssemblyMetadata.Create(moduleMetadata);
-                    references.Add(assemblyMetadata.GetReference());
+                    using (var stream = File.OpenRead(lib.ResolvedPath))
+                    {
+                        var moduleMetadata = ModuleMetadata.CreateFromStream(stream, PEStreamOptions.PrefetchMetadata);
+                        assemblyMetadata = AssemblyMetadata.Create(moduleMetadata);
+                        references.Add(assemblyMetadata.GetReference());
+                    }
+                }
+                catch (Exception ex) 
+                {
+                    if(throwOnError) 
+                    {
+                        throw ex;
+                    }
+                    continue;
```

Comment:
Right now, the whole error message experience is pretty crude for the user. 
If compilation fails, it would show the compilation errors to the user. 
Types of exception expected here: 
FileNotFoundException, NotSupportedException, ArgumentException, ArgumentOutOfRangeException, BadImageFormatException, IOException, ArgumentNullException. 

Will change to catch these specific exceptions. 

Also I think currently the debug logging is not great. I will open a new issue for debug logging this and other exceptions/ errors.